tests/core/uc20-recovery/task.yaml

```diff
@@ -0,0 +1,85 @@
+summary: verify booting into recovery mode on UC20 systems
+
+systems: [ ubuntu-core-20-* ]
+
+restore: |
+  rm -f /writable/systems.json.run
+  rm -f /writable/systems.label
+  rm -f system-info systems.json
+
+  snap remove http jq
+
+debug: |
+  if [ -e systems.json ]; then
+      cat systems.json
+  fi
+
+execute: |
+  if [[ "$SPREAD_REBOOT" == "0" ]]; then
+      echo "In run mode"
+
+      snap install jq
+      # devmode as the snap does not have snapd-control
+      snap install test-snapd-curl --devmode --edge
+
+      MATCH 'snapd_recovery_mode=run' < /proc/cmdline
+      # verify we are in run mode via the API
+      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/system-info > system-info
```

Comment:
hmm the http snap seems to work for me fine with this:

```
$ sudo http -p b snapd:///v2/systems | sudo jq -r '.result'
{
  "systems": [
    {
      "current": true,
      "label": "20200512",
      "model": {
        "model": "ubuntu-core-20-amd64-dangerous",
        "brand-id": "canonical",
        "display-name": "ubuntu-core-20-amd64-dangerous"
      },
      "brand": {
        "id": "canonical",
        "username": "canonical",
        "display-name": "Canonical",
        "validation": "verified"
      },
      "actions": [
        {
          "title": "Reinstall",
          "mode": "install"
        },
        {
          "title": "Recover",
          "mode": "recover"
        },
        {
          "title": "Run normally",
          "mode": "run"
        }
      ]
    }
  ]
}
```

(that was without this PR, so no system-mode in the output)

Reply:
```
diff --git a/tests/core/uc20-recovery/task.yaml b/tests/core/uc20-recovery/task.yaml
index c91d8c3d7e..bbed751a6e 100644
--- a/tests/core/uc20-recovery/task.yaml
+++ b/tests/core/uc20-recovery/task.yaml
@@ -21,10 +21,12 @@ execute: |
       snap install jq
       # devmode as the snap does not have snapd-control
       snap install test-snapd-curl --devmode --edge
+      snap install http
 
       MATCH 'snapd_recovery_mode=run' < /proc/cmdline
       # verify we are in run mode via the API
-      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/system-info > system-info
+      # test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/system-info > system-info
+      snap run http snapd:///v2/system-info > system-info
       jq -r '.result["system-mode"]' < system-info | MATCH 'run'
 
       echo "Obtain available systems"
```

When I use `http` in the test is what I get:
```
+ snap run http snapd:///v2/system-info
+ MATCH run
+ jq -r '.result["system-mode"]'
grep error: pattern not found, got:
null
-----
```
The response is:
```
{"type":"error","status-code":405,"status":"Method Not Allowed","result":{"message":"method \"POST\" not allowed"}}
```
The last thing that appears in snapd journal:
```
May 12 18:32:48 ubuntu snapd[3321]: taskrunner.go:437: DEBUG: Running task 169 on Do: Run configure hook of "http" snap if present
May 12 18:32:48 ubuntu snapd[3321]: taskrunner.go:437: DEBUG: Running task 170 on Do: Run health check of "http" snap
May 12 18:32:48 ubuntu snapd[3321]: daemon.go:313: DEBUG: pid=3675;uid=0;socket=/run/snapd.socket; GET /v2/snaps?snaps=http 727.629µs 200
^^^ the end of snap install http
May 12 18:32:50 ubuntu snapd[3321]: daemon.go:313: DEBUG: pid=3779;uid=0;socket=/run/snapd.socket; POST /v2/system-info 66.528µs 405
```
Of course it works int he debug shell and does a GET request.

Reply:
oh that's real weird, but oh well the test-snapd-curl snap is fine for this purpose I think, but it would be nice to know why http snap behaves oddly here